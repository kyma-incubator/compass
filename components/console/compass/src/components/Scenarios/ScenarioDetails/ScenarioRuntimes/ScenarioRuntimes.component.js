import React from 'react';
import PropTypes from 'prop-types';
import { GenericList } from 'react-shared';
import { useMutation } from '@apollo/react-hooks';

import AssignEntityToScenarioModal from '../shared/AssignEntityToScenarioModal/AssignRuntimesToScenarioModal.container';
import unassignScenarioHandler from '../shared/unassignScenarioHandler';
import { SEND_NOTIFICATION } from '../../../../gql';

ScenarioRuntimes.propTypes = {
  scenarioName: PropTypes.string.isRequired,
  getRuntimesForScenario: PropTypes.object.isRequired,
  setRuntimeScenarios: PropTypes.func.isRequired,
  deleteRuntimeScenarios: PropTypes.func.isRequired,
};

export default function ScenarioRuntimes({
  scenarioName,
  getRuntimesForScenario,
  setRuntimeScenarios,
  deleteRuntimeScenarios,
}) {
  const [sendNotification] = useMutation(SEND_NOTIFICATION);

  if (getRuntimesForScenario.loading) {
    return <p>Loading...</p>;
  }
  if (getRuntimesForScenario.error) {
    return `Error! ${getRuntimesForScenario.error.message}`;
  }

  const showSuccessNotification = runtimeName => {
    sendNotification({
      variables: {
        content: `Unassigned "${runtimeName}" from ${scenarioName}.`,
        title: runtimeName,
        color: '#359c46',
        icon: 'accept',
        instanceName: scenarioName,
      },
    });
  };

  const actions = [
    {
      name: 'Unassign',
      handler: async runtime => {
        await unassignScenarioHandler(
          runtime.name,
          runtime.id,
          runtime.labels.scenarios,
          setRuntimeScenarios,
          deleteRuntimeScenarios,
          scenarioName,
          async () => {
            showSuccessNotification(runtime.name);
            await getRuntimesForScenario.refetch();
          },
        );
      },
    },
  ];

  const assignedRuntimes = getRuntimesForScenario.runtimes.data;

  const extraHeaderContent = (
    <AssignEntityToScenarioModal
      originalEntities={assignedRuntimes}
      entitiesForScenarioRefetchFn={getRuntimesForScenario.refetch}
    />
  );

  return (
    <GenericList
      extraHeaderContent={extraHeaderContent}
      title="Runtimes"
      notFoundMessage="No Runtimes for this Scenario"
      entries={assignedRuntimes}
      headerRenderer={() => ['Name']}
      actions={actions}
      rowRenderer={runtime => [runtime.name]}
      showSearchField={false}
    />
  );
}
