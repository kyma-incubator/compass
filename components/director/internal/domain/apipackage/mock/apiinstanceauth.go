package mock

import (
	"time"

	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
)

func FixAPIInstanceAuth(id string, condition graphql.APIInstanceAuthStatusCondition) *graphql.APIInstanceAuth {
	var reason string
	var auth *graphql.Auth

	switch condition {

	case graphql.APIInstanceAuthStatusConditionSucceeded:
		reason = "CredentialsProvided"
		auth = &graphql.Auth{
			Credential: graphql.BasicCredentialData{
				Username: "username",
				Password: "password",
			},
		}
	case graphql.APIInstanceAuthStatusConditionFailed:
		reason = "CredentialsNotProvided"
	}

	return &graphql.APIInstanceAuth{
		ID:      id,
		Context: nil,
		Auth:    auth,
		Status: &graphql.APIInstanceAuthStatus{
			Condition: condition,
			Timestamp: graphql.Timestamp(time.Now()),
			Message:   "Message",
			Reason:    reason,
		},
	}

}
