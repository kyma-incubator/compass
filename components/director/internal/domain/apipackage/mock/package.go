package mock

import "github.com/kyma-incubator/compass/components/director/pkg/graphql"

func FixPackage() *graphql.PackageDefinition {
	desc := "Lorem ipsum"
	jsonSchema := `{
  "definitions": {}, 
  "$schema": "http://json-schema.org/draft-07/schema#", 
  "$id": "http://example.com/root.json", 
  "type": "object", 
  "title": "The Root Schema", 
  "description": "An explanation about the purpose of this instance.", 
  "readOnly": true, 
  "writeOnly": false, 
  "required": [
    "checked", 
    "dimensions", 
    "id", 
    "name", 
    "price", 
    "tags"
  ], 
  "properties": {
    "checked": {
      "$id": "#/properties/checked", 
      "type": "boolean", 
      "title": "The Checked Schema", 
      "description": "An explanation about the purpose of this instance.", 
      "default": false, 
      "examples": [
        false
      ], 
      "readOnly": true, 
      "writeOnly": false
    }, 
    "dimensions": {
      "$id": "#/properties/dimensions", 
      "type": "object", 
      "title": "The Dimensions Schema", 
      "description": "An explanation about the purpose of this instance.", 
      "readOnly": true, 
      "writeOnly": false, 
      "required": [
        "width", 
        "height"
      ], 
      "properties": {
        "width": {
          "$id": "#/properties/dimensions/properties/width", 
          "type": "integer", 
          "title": "The Width Schema", 
          "description": "An explanation about the purpose of this instance.", 
          "default": 0, 
          "examples": [
            5
          ], 
          "readOnly": true, 
          "writeOnly": false
        }, 
        "height": {
          "$id": "#/properties/dimensions/properties/height", 
          "type": "integer", 
          "title": "The Height Schema", 
          "description": "An explanation about the purpose of this instance.", 
          "default": 0, 
          "examples": [
            10
          ], 
          "readOnly": true, 
          "writeOnly": false
        }
      }
    }, 
    "id": {
      "$id": "#/properties/id", 
      "type": "integer", 
      "title": "The Id Schema", 
      "description": "An explanation about the purpose of this instance.", 
      "default": 0, 
      "examples": [
        1
      ], 
      "readOnly": true, 
      "writeOnly": false
    }, 
    "name": {
      "$id": "#/properties/name", 
      "type": "string", 
      "title": "The Name Schema", 
      "description": "An explanation about the purpose of this instance.", 
      "default": "", 
      "examples": [
        "A green door"
      ], 
      "readOnly": true, 
      "writeOnly": false, 
      "pattern": "^(.*)$"
    }, 
    "price": {
      "$id": "#/properties/price", 
      "type": "number", 
      "title": "The Price Schema", 
      "description": "An explanation about the purpose of this instance.", 
      "default": 0.0, 
      "examples": [
        12.5
      ], 
      "readOnly": true, 
      "writeOnly": false
    }, 
    "tags": {
      "$id": "#/properties/tags", 
      "type": "array", 
      "title": "The Tags Schema", 
      "description": "An explanation about the purpose of this instance.", 
      "readOnly": true, 
      "writeOnly": false, 
      "items": {
        "$id": "#/properties/tags/items", 
        "type": "string", 
        "title": "The Items Schema", 
        "description": "An explanation about the purpose of this instance.", 
        "default": "", 
        "examples": [
          "home", 
          "green"
        ], 
        "readOnly": true, 	
        "writeOnly": false, 
        "pattern": "^(.*)$"
      }
    }
  }
}`
	gqlJSONSchema := graphql.JSONSchema(jsonSchema)

	return &graphql.PackageDefinition{
		ID:                    "a6a64619-aa28-4d5d-bbe8-86ab4b817abf",
		Name:                  "Lorem ipsum dolores sit amet",
		Description:           &desc,
		AuthRequestJSONSchema: &gqlJSONSchema,
		DefaultAuth: &graphql.Auth{
			Credential: graphql.BasicCredentialData{
				Username: "foo",
				Password: "bar",
			},
		},
	}
}
