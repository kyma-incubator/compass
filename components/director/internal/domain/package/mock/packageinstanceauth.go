package mock

import (
	"time"

	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
)

func FixPackageInstanceAuth(id string, condition graphql.PackageInstanceAuthStatusCondition) *graphql.PackageInstanceAuth {
	var reason string
	var auth *graphql.Auth

	switch condition {

	case graphql.PackageInstanceAuthStatusConditionSucceeded:
		reason = "CredentialsProvided"
		auth = &graphql.Auth{
			Credential: graphql.BasicCredentialData{
				Username: "username",
				Password: "password",
			},
		}
	case graphql.PackageInstanceAuthStatusConditionFailed:
		reason = "CredentialsNotProvided"
	}

	return &graphql.PackageInstanceAuth{
		ID:      id,
		Context: nil,
		Auth:    auth,
		Status: &graphql.PackageInstanceAuthStatus{
			Condition: condition,
			Timestamp: graphql.Timestamp(time.Now()),
			Message:   "Message",
			Reason:    reason,
		},
	}

}
