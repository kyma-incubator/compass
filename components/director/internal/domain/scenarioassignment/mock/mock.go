package mock

import "github.com/kyma-incubator/compass/components/director/pkg/graphql"

func FixAssignmentForScenario(scenarioName string) *graphql.AutomaticScenarioAssignment {
	selector := &graphql.Label{
		Key:   "selector",
		Value: "dummy-value",
	}
	return FixAssignmentForScenarioWithSelector(scenarioName, selector)
}

func FixAssignmentForScenarioWithSelector(scenarioName string, selector *graphql.Label) *graphql.AutomaticScenarioAssignment {
	return &graphql.AutomaticScenarioAssignment{
		ScenarioName: scenarioName,
		Selector:     selector,
	}
}
