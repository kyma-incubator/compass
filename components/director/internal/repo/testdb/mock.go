package testdb

import (
	"testing"

	"github.com/DATA-DOG/go-sqlmock"
	"github.com/jmoiron/sqlx"
	"github.com/stretchr/testify/require"
)

// MockDatabase returns a new *sqlx.DB mock alongside with a wrapper providing easier interface for asserting expectations.
func MockDatabase(t *testing.T) (*sqlx.DB, DBMock) {
	sqlDB, sqlMock, err := sqlmock.New()
	require.NoError(t, err)

	sqlxDB := sqlx.NewDb(sqlDB, "sqlmock")

	return sqlxDB, &sqlMockWithAssertions{sqlMock}
}

// DBMock represents a wrapper providing easier interface for asserting expectations.
type DBMock interface {
	sqlmock.Sqlmock
	AssertExpectations(t *testing.T)
}

type sqlMockWithAssertions struct {
	sqlmock.Sqlmock
}

// AssertExpectations asserts that all the expectations to the mock were met.
func (s *sqlMockWithAssertions) AssertExpectations(t *testing.T) {
	err := s.ExpectationsWereMet()
	require.NoError(t, err)
}
