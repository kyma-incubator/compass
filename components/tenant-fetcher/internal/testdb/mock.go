package testdb

import (
	"testing"

	"github.com/DATA-DOG/go-sqlmock"
	"github.com/jmoiron/sqlx"
	"github.com/stretchr/testify/require"
)

func MockDatabase(t *testing.T) (*sqlx.DB, DBMock) {
	sqlDB, sqlMock, err := sqlmock.New()
	require.NoError(t, err)

	sqlxDB := sqlx.NewDb(sqlDB, "sqlmock")

	return sqlxDB, &sqlMockWithAssertions{sqlMock}
}

type DBMock interface {
	sqlmock.Sqlmock
	AssertExpectations(t *testing.T)
}

type sqlMockWithAssertions struct {
	sqlmock.Sqlmock
}

func (s *sqlMockWithAssertions) AssertExpectations(t *testing.T) {
	err := s.ExpectationsWereMet()
	require.NoError(t, err)
}
