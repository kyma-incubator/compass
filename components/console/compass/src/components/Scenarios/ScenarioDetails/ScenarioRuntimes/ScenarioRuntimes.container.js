import { graphql } from 'react-apollo';
import { compose } from 'recompose';
import { fromRenderProps } from 'recompose';
import {
  GET_RUNTIMES_FOR_SCENARIO,
  SET_RUNTIME_SCENARIOS,
  createEqualityQuery,
  DELETE_RUNTIME_SCENARIOS_LABEL,
} from '../../gql';
import { SEND_NOTIFICATION } from '../../../../gql';

import ScenarioRuntimes from './ScenarioRuntimes.component';
import ScenarioNameContext from './../ScenarioNameContext';

export default compose(
  fromRenderProps(ScenarioNameContext.Consumer, scenarioName => ({
    scenarioName,
  })),
  graphql(SET_RUNTIME_SCENARIOS, {
    props: ({ mutate }) => ({
      setRuntimeScenarios: async variables => await mutate(variables),
    }),
  }),
  graphql(DELETE_RUNTIME_SCENARIOS_LABEL, {
    props: ({ mutate }) => ({
      deleteRuntimeScenarios: async id =>
        await mutate({
          variables: { id },
        }),
    }),
  }),
  graphql(GET_RUNTIMES_FOR_SCENARIO, {
    name: 'getRuntimesForScenario',
    options: ({ scenarioName }) => {
      const filter = {
        key: 'scenarios',
        query: createEqualityQuery(scenarioName),
      };
      return {
        variables: {
          filter: [filter],
        },
      };
    },
  }),
  graphql(SEND_NOTIFICATION, {
    name: 'sendNotification',
  }),
)(ScenarioRuntimes);
