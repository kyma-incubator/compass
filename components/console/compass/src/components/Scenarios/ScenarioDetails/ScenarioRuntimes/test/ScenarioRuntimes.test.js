import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ScenarioRuntimes from '../ScenarioRuntimes.component';
import { responseMock } from './mock';
jest.mock('react-shared', () => ({
  GenericList: function GenericListMocked(props) {
    return 'generic-list-mocked-content';
  },
}));

describe('ScenarioRuntimes', () => {
  it('Renders with minimal props', () => {
    const component = shallow(
      <ScenarioRuntimes
        scenarioName="scenario name"
        getRuntimesForScenario={responseMock}
        deleteRuntimeScenarios={() => {}}
        setRuntimeScenarios={() => {}}
        sendNotification={() => {}}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
