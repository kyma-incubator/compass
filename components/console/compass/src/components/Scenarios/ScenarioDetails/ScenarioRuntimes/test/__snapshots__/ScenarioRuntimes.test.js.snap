// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScenarioRuntimes Renders with minimal props 1`] = `
<GenericListMocked
  actions={
    Array [
      Object {
        "handler": [Function],
        "name": "Unassign",
      },
    ]
  }
  entries={
    Array [
      Object {
        "id": "id 1",
        "labels": Object {},
        "name": "Runtime 1",
      },
    ]
  }
  extraHeaderContent={
    <Apollo(Apollo(Apollo(withProps(AssignEntityToScenarioModal))))
      originalEntities={
        Array [
          Object {
            "id": "id 1",
            "labels": Object {},
            "name": "Runtime 1",
          },
        ]
      }
    />
  }
  headerRenderer={[Function]}
  notFoundMessage="No Runtimes for this Scenario"
  rowRenderer={[Function]}
  showSearchField={false}
  title="Runtimes"
/>
`;
